dotnet/src/SemanticKernel.Abstractions/AI/PromptExecutionSettings.cs

```diff
@@ -39,5 +40,15 @@ public class PromptExecutionSettings
     /// Avoid using this property if possible. Instead, use one of the classes that extends <see cref="PromptExecutionSettings"/>.
     /// </remarks>
     [JsonExtensionData]
-    public Dictionary<string, object>? ExtensionData { get; set; }
+    public virtual Dictionary<string, object>? ExtensionData { get; set; }
```

Comment:
_Technically_ making something virtual is a breaking change. However, at this point in SK's adoption, it's ok. The impact is existing callers in compiled binaries will be using call rather than callvirt, which means they won't see any overrides and will continue to use the base implementation.